lib/rules/links/linkchecker.js

```diff
@@ -47,28 +47,29 @@ exports.check = async function (sr, done) {
     const browser = await puppeteer.launch();
     const page = await browser.newPage();
 
-    const dts = sr.extractHeaders(
-        sr.jsDocument.querySelector('body div.head dl')
-    );
-    // thisLink is used as base url. Every other resources should use in same folder as base. e.g.
-    // url: https://www.w3.org/TR/css3-conditional/
-    // image: https://www.w3.org/TR/2020/css3-conditional/image/img.jpg
-    const thisLink = dts && dts.This && dts.This.dd.querySelector('a').href;
-    if (!thisLink) {
+    // sr.url is used as base url. Every other resources should use in same folder as base. e.g.
+    // - spec doc: https://www.w3.org/TR/2021/WD-pubrules-20210401/
+    // - image (pass): https://www.w3.org/TR/2021/WD-pubrules-20210401/images/sample.png
+    // - image (pass): https://www.w3.org/TR/2021/WD-pubrules-20210401/sample.png
+    // - image (error): https://w3c.github.io/pubrules/sample.png
+
+    if (!sr.url) {
+        await browser.close();
```

Comment:
As you mentioned in our last call, you can move the launch of puppeteer after that `if`